### DISCUSSION TITLE
Undo and redo buttons

### DISCUSSION THREAD

**Discussion by CallerOnLineOne** [GrapesJS/grapesjs#4311] [Q&A]
How do I add undo and redo button. I see the code for the undo_manager but I can't figure out how or where to addthe actual undo redo buttons to call it

**Comment by artf** [comment]
If you want to reuse the default panels module, it might be something like this
```js
editor.Panels.addButton('options', [
    { id: 'undo', className: 'fa fa-undo', command: 'core:undo' },
    { id: 'redo', className: 'fa fa-repeat', command: 'core:redo' }
]);
```

**Reply by CallerOnLineOne** [reply]
Thank you! I had them in there but they weren't functioning. I had the wrong value for the commands
